src/openapi-generator.ts

```diff
@@ -783,59 +783,46 @@ export class OpenAPIGenerator {
     return zodSchema.isOptional();
   }
 
+  private requiredKeysOf(objectSchema: ZodObject<ZodRawShape>) {
+    return Object.entries(objectSchema._def.shape())
+      .filter(([_key, type]) => !this.isOptionalSchema(type))
+      .map(([key, _type]) => key);
+  }
+
   private toOpenAPIObjectSchema(
     zodSchema: ZodObject<ZodRawShape>,
     isNullable: boolean
   ): SchemaObject {
     const extendedFrom = zodSchema._def.openapi?.extendedFrom;
 
-    const propTypes = zodSchema._def.shape();
-    const unknownKeysOption = zodSchema._unknownKeys as UnknownKeysParam;
-
-    const requiredProperties = Object.entries(propTypes)
-      .filter(([_key, type]) => !this.isOptionalSchema(type))
-      .map(([key, _type]) => key);
-
-    const schemaProperties = mapValues(propTypes, propSchema =>
-      this.generateInnerSchema(propSchema)
-    );
-
-    let alreadyRegistered: string[] = [];
-    let alreadyRequired: string[] = [];
-
-    if (extendedFrom) {
-      const registeredSchema = this.schemaRefs[extendedFrom];
-
-      if (!registeredSchema) {
```

Comment:
We don't need it because we now only have `extendedFrom` if we're extending from a defined schema. And we're not using that schema from `schemaRefs` so we don't need to double-check.